cloud_chatops/src/pr_reminder.py

```diff
@@ -0,0 +1,205 @@
+        self.client = WebClient(token=get_token("SLACK_BOT_TOKEN"))
+        self.slack_ids = get_user_map()
+        self.prs = GetGitHubPRs(get_repos(), "stfc").run()
+        self.channel = "temp-chatops"
+
+    def run(self, mention, channel=None) -> None:
+        """
+        This method calls the functions to post the reminder message and subsequent PR messages.
+        :param channel: Changes the channel to post the messages to.
+        :param mention: To mention the users in Slack or just post their name.
+        """
+        if channel:
+            self.channel = channel
+        reminder_message = self.post_reminder_message()
+        self.post_thread_messages(self.prs, reminder_message, mention)
+
+    def post_reminder_message(self) -> WebClient.chat_postMessage:
+        """
+        This method posts the main reminder message to start the thread if PR notifications.
+        :return: The reminder message return object
+        """
+        reminder = self.client.chat_postMessage(
+            channel=self.channel,
+            text="Here are the outstanding PRs as of today:",
+        )
+        return reminder
+
+    def post_thread_messages(
+        self,
+        prs: Dict[str, List],
+        reminder_message: WebClient.chat_postMessage,
+        mention: bool,
+    ) -> None:
+        """
+        This method iterates through each PR and calls the post method for them.
+        :param mention: To mention the users or not
+        :param prs: A list of PRs from GitHub
+        :param reminder_message: The reminder message object
+        """
+        prs_found = False
+        for repo in prs.values():
+            for pr in repo:
+                info = {
+                    "pr_title": f"{pr['title']} #{pr['number']}",
+                    "user": pr["user"]["login"],
+                    "url": pr["html_url"],
+                    "created_at": pr["created_at"],
+                    "channel": reminder_message.data["channel"],
+                    "thread_ts": reminder_message.data["ts"],
+                    "mention": mention,
+                    "draft": pr["draft"],
+                }
+                prs_found = True
+                checked_info = self.check_pr(info)
+                self.send_thread(**checked_info)
+
+        if not prs_found:
+            self.send_no_prs(reminder_message)
+
+    def send_no_prs(self, reminder: WebClient.chat_postMessage) -> None:
+        """
+        This method sends a message to the user that they have no PRs open.
+        This method is only called if no other PRs have been mentioned.
+        :param reminder: The thread message to send under.
+        """
+        self.client.chat_postMessage(
+            text="There are no outstanding pull requests open.",
+            channel=reminder.data["channel"],
+            thread_ts=reminder.data["ts"],
+            unfurl_links=False,
+        )
+
+    def check_pr(self, info: Dict) -> Dict:
```

Comment:
this class is also getting too big - you have this doing 2 things, validating the PR and posting it - which should be two separate classes same as above

Reply:
You should break the class up into 

`PRMessageBuilder` - which checks various collected PRs, validates and builds a message to send
`PRMessagePoster` - which takes a list of PR messages and posts them as a single slack message.